lib/std/mem.zig

```diff
@@ -3262,7 +3262,7 @@ test "indexOfMax" {
 /// Finds the indices of the smallest and largest number in a slice. O(n).
 /// Returns an anonymous struct with the fields `index_min` and `index_max`.
 /// `slice` must not be empty.
-pub fn indexOfMinMax(comptime T: type, slice: []const T) struct { index_min: usize, index_max: usize } {
+pub fn indexOfMinMax(comptime T: type, slice: []const T) IndexOfMinMaxResult {
```

Comment:
This seems suspiciously like a regression. Does `.{ .index_min = 0, .index_max = 6 }` not work for the test cases?

Reply:
Doesn't work for the same reason as `test/behavior/cast.zig`. You have to use reflection and get the type of the actual value and explicitly coerce to it, which is very inconvenient.

```zig
pub fn indexOfMinMax(comptime T: type, slice: []const T) struct { index_min: usize, index_max: usize } {
    // ...
}

test "indexOfMinMax" {
    const actual = indexOfMinMax(u8, "abcdefg");
    try testing.expectEqual(@as(@TypeOf(actual), .{ .index_min = 0, .index_max = 6 }), actual);
}
```